packages/graphql/src/translate/queryAST/factory/OperationFactory.ts

```diff
@@ -287,7 +287,7 @@ export class OperationsFactory {
         } else {
             // if typename is allowed and therefore _on is disabled we can compute only the shared filter without recomputing the filters for each concrete entity
             // if typename filters are allowed we are getting rid of the _on and the implicit typename filter.
-            const typenameFilterAllowed = this.experimental && isInterfaceEntity(entity);
+            const typenameFilterAllowed = isInterfaceEntity(entity);
```

Comment:
I will change the name from `typenameFilterAllowed` to `isInterface` as by removing the experimental flag any interface supports typename filters.